### ISSUE TITLE
Slip through the CREATING phase

### ISSUE THREAD

**Issue by utam0k** [gitpod-io/gitpod#15406]
### Bug description

This issue is blocker for our deployment.

Our loadgen sometimes can't pass with the config for production. As far as my experience, I can usually launch about 98 workspaces, but 100 is tough.
Especially those situations that are harder

a. The first time(Probably related to autoscale)
b. When I launch 100 workspaces and then launch adding 100 while deleting the previous 100 workspaces

I tried to case a and got the error on `ephemeral-jt` when starting the workspaces.
```text
WARN[0665] workspace 84f3fafb-ed6d-4be3-b3e3-46fffb22b62b is in phase STOPPED 
WARN[0665] workspace 6384256f-7a01-410b-9acf-0f5fa376b06b is in phase STOPPED 
WARN[0665] workspace 586e97fb-a4a1-4c82-92b2-8bcd7c03e74c is in phase STOPPED 
WARN[0665] workspace 28bca74b-8a50-473d-86f0-90ac2e5fc781 is in phase STOPPED 
WARN[0665] workspace 504ce8a3-1990-498a-a88b-3a79cafb2829 is in phase STOPPED 
WARN[0665] workspace d1cd741e-6738-4fe4-95a1-b89d3e472d93 is in phase STOPPED 
WARN[0665] workspace 6c13950f-ba33-41d9-800a-4142a2a3db9c is in phase STOPPED 
WARN[0665] workspace b68d3e45-369f-4d3a-8d15-eaf8d80ef644 is in phase STOPPED 
WARN[0665] workspace 04b74623-123d-4ae2-ba78-2a5857d4a5bc is in phase STOPPED 
WARN[0665] workspace 038d77f8-988f-40bd-8fb6-bd749792fb3d is in phase STOPPED 
```

I dive into gcp logging and trace a bit. I showed one sample workspace, `84f3fafb-ed6d-4be3-b3e3-46fffb22b62b`.

#### gcp logging

[Go to the log of this instance](https://cloudlogging.app.goo.gl/rbyBRAdZS1Yubc6K6)
Unfortunately, this log only complained about `workspace failed`.

#### trace

[Go to the trace of this instance](https://jaeger-us-prod.workspace-preview.gitpod-io-dev.com/search?end=1671434953207000&limit=20&lookback=1h&maxDuration&minDuration&service=ws-manager&start=1671431353207000&tags=%7B%22instanceId%22%3A%2284f3fafb-ed6d-4be3-b3e3-46fffb22b62b%22%7D)

I found some curious traces.

- [trace A](https://jaeger-us-prod.workspace-preview.gitpod-io-dev.com/trace/64bd94b049a80055)
- [trace B](https://jaeger-us-prod.workspace-preview.gitpod-io-dev.com/trace/27940e666153decc)

From traces, we can find that the workspace became a stop phase from the running phase for some reason.
![image](https://user-images.githubusercontent.com/13010913/208376858-4f125349-3a61-474e-9734-5dcf47c4219e.png)
![image](https://user-images.githubusercontent.com/13010913/208377228-685d246b-4af0-42f6-9749-9770f1a9403a.png)

In other words, After the workspace reaches the running phase, the workspace goes to the stop phase for some reason.

### Steps to reproduce

Run loadgen for a epehmeral cluster

**Comment by kylos101** [comment]
@utam0k what is the problem, that some workspaces do not start, or, they start but it takes too long and is considered a timeout? May I ask you to add the actual error, and update the title and desc to reflect if it is a timeout, or something else causing failure?

**Comment by utam0k** [comment]
> @utam0k what is the problem, that some workspaces do not start, or, they start but it takes too long and is considered a timeout? May I ask you to add the actual error, and update the title and desc to reflect if it is a timeout, or something else causing failure?

AFAI, some workspaces got failed and never woke up. Should I create an ephemeral cluster and got the error?
FYI: This issue came up from our retrospective

**Comment by utam0k** [comment]
@kylos101 I have updated. This probably happened on the production

**Comment by kylos101** [comment]
@utam0k given these [logs](https://cloudlogging.app.goo.gl/4whxKejbB3PpJM579), I see [this error](https://console.cloud.google.com/logs/query;cursorTimestamp=2022-12-19T07:08:36.551096689Z;query=resource.labels.cluster_name%3D%22ws-ephemeral-jt%22%0A%2284f3fafb-ed6d-4be3-b3e3-46fffb22b62b%22%0A-resource.labels.container_name%3D%22calico-node%22%0Atimestamp%3D%222022-12-19T07:08:35.693947918Z%22%0AinsertId%3D%22iybg3pievqwu9mxe%22;summaryFields=jsonPayload%252FworkspaceId,jsonPayload%252FinstanceId:false:32:beginning;timeRange=2022-12-19T07:00:00.000Z%2F2022-12-19T07:18:00.000Z?project=workspace-clusters), seems like the workspace could not connect to ws-daemon after two minutes, and then failed.

Was the node overwhelmed with stress-ng workspaces?

**Comment by utam0k** [comment]
> Was the node overwhelmed with stress-ng workspaces? Was the node starved for resources?

Of course, my load test has `stree-ns` workspaces, but I'm not sure. But I think it is necessary to handle the connection to ws-daemon with fewer resources to make it succeed or not land a new workspace on that node.

**Comment by kylos101** [comment]
With us80, we were able to start the initial 100, and the second 100 did not all start while the first 100 stopped, 86 started, 14 failed.

The 14 failed all had the same trend:  https://cloudlogging.app.goo.gl/FRKCBY1MMJbzoMQN9

![image](https://user-images.githubusercontent.com/1272076/208786007-6d53b48b-3530-4a0e-bffa-543b7f074830.png)

**Comment by utam0k** [comment]
`Two minutes` is the same as what we set the timeout for connecting to ws-daemon from a workspace(ring0)
https://github.com/gitpod-io/gitpod/blob/a33857ed6d64f50a1c597b2d0a7e81adf13f780c/components/workspacekit/cmd/rings.go#L81-L88

**Comment by kylos101** [comment]
> `Two minutes` is the same as what we set the timeout for connecting to ws-daemon from a workspace(ring0)
> 
> https://github.com/gitpod-io/gitpod/blob/a33857ed6d64f50a1c597b2d0a7e81adf13f780c/components/workspacekit/cmd/rings.go#L81-L88

Indeed, I thought it was familiar

**Comment by utam0k** [comment]
I picked it as an on-caller job.

**Comment by utam0k** [comment]
I found that, for some reason, the workspaces that got failed started slipping through the CREATING phase. In other words, they don't run hooks, including the task for starting the IWS server.
![image](https://user-images.githubusercontent.com/13010913/209080392-e86590fd-0082-461f-83cf-a6b236ebd519.png)
